core/src/apps/management/apply_settings.py

```diff
@@ -10,11 +10,32 @@
 if TYPE_CHECKING:
     from trezor.enums import SafetyCheckLevel
     from trezor.messages import ApplySettings, Success
+    from trezor.ui import ProgressLayout
 
 
 BRT_PROTECT_CALL = ButtonRequestType.ProtectCall  # CACHE
 
 
+async def _load_homescreen(length: int) -> bytes:
+    from trezor import utils, workflow
+    from trezor.ui.layouts.progress import progress
+
+    from apps.common import chunked
+
+    loader: ProgressLayout | None = None
+
+    def report(value: int) -> None:
+        nonlocal loader
+        if loader is None:
+            workflow.close_others()
+            loader = progress(TR.progress__please_wait)
+        loader.report(value)
+
+    buf = utils.empty_bytearray(length)
```

Comment:
Should we limit `length` (to avoid OOMs)?

Reply:
well, it's safe to do it like this, the worst an attacker can do is get back a crash if they send an image size that's too much
but it would be nicer to pre-limit `length` to allowable homescreen lengths anyway